### ISSUE TITLE
Textarea as custom field

### ISSUE THREAD

**Issue by omtr** [SiimKirjanen/seatreg#123]
the GDPR requires to inform the user about certain aspects of the data processing, e.g.

- what data is process
- how the data is processed
- what are the rights of the data subject (right to be informed, right to delete etc.)

In order to make Seatreg more GDPR compliant, it would be great to have a textarea that can be filled with GDPR info and a reference link to the website'S privacy statement.

**Comment by SiimKirjanen** [comment]
Not sure if its the responsibility of this plugin. For GDPR info you can create a normal WordPress page and fill in the required info. And if you are using SeatReg shortcode then you can just add a link to this GDPR page under the shortcode.

**Comment by omtr** [comment]
Hi Siim,
according to the GDPR, "where data is obtained directly, the person must be immediately informed, meaning at the time the data is obtained." Source: [https://gdpr-info.eu/issues/right-to-be-informed/]
That's why it is a recommended practice to add privacy statement under each web form or - as you already mentioned - to add a link to the privacy statement.
Adding a link below the shortcode is a feasible way to do it, but it does not really look good and is not part of the form itself. The best pleace would be a disclaimer statement that could be placed next to the form submit button.

**Comment by SiimKirjanen** [comment]
Okay. Let's implement URL field to options. So if its filled then show the link below the booking submit form.

**Comment by kalle-1989** [comment]
@omtr I had the same issue with displaying some GDPR text above the "Book" button. 
@SiimKirjanen I did some prototyping for my project  (https://github.com/SiimKirjanen/seatreg/compare/master...kalle-1989:seatreg:master-1.3.8.1-mod-custom-footer-text) and added a new field to the config page where a user could add custom HTML text:
![grafik](https://github.com/SiimKirjanen/seatreg/assets/46853637/625d923a-93b2-4106-b18c-0dbdfa1445a9)
This text is then displayed above the confirm button:
![grafik](https://github.com/SiimKirjanen/seatreg/assets/46853637/d0b5f724-d52d-4f8a-91f7-626c40c59e8c)

@SiimKirjanen Maybe you want to adopt the idea. I had some problems with displaying the linebreaks ("\n") when reading the HTML string from the database. Not sure if my solution is the best

**Comment by SiimKirjanen** [comment]
@kalle-1989 I think the better solution is to create a separate WordPress page for GDPR. Because I think the GDPR text can be very long and the booking form modal does not have room for it (we also need to display custom fields there if created).
The idea was to create a field for options page where you can provide the link to GDPR page. And in booking modal add a link below the Book button. So users can click on it and read GDPR info on separate page.

**Comment by SiimKirjanen** [comment]
Or we can open up a separate modal that displays the GDPR text 🤔 
Not sure if there are any requirements where the GDPR info is displayed.

**Comment by SiimKirjanen** [comment]
I think the separate WordPress page is better as user has more control over it.

**Comment by omtr** [comment]
The only requirement is that it is easily accessible and in an easy to understand language. A simple link is totally sufficient, and I think it would be better to link to an existing privacy statement for the sake of easier maintenance.

**Comment by kalle-1989** [comment]
> @kalle-1989 I think the better solution is to create a separate WordPress page for GDPR. Because I think the GDPR text can be very long and the booking form modal does not have room for it (we also need to display custom fields there if created). The idea was to create a field for options page where you can provide the link to GDPR page. And in booking modal add a link below the Book button. So users can click on it and read GDPR info on separate page.

Sorry. I think I was not clear enough with my requirement. I just want to give the user the ability to add (short) custom text which could be used for example for a link to GDPR.  I don't want to display the full GDPR. In my case, it would look like this example.
![grafik](https://github.com/SiimKirjanen/seatreg/assets/46853637/0e416103-e461-44fd-9faa-a803b3083751)
![grafik](https://github.com/SiimKirjanen/seatreg/assets/46853637/9df8db5c-333a-493b-b70f-b4b5f5eb13c7)

**Comment by kalle-1989** [comment]
Another option may could be to allow HTML code for custom field. But this is currently blocked by a character check.
![grafik](https://github.com/SiimKirjanen/seatreg/assets/46853637/78d8e155-9f04-4917-89bf-d1e206a634df)

**Comment by SiimKirjanen** [comment]
@kalle-1989  Oh sorry I did not notice that you can actually add links with the text editor. Now I like the idea. Can you create a pull request. I will review it.
